gossip-transport-http/src/main/java/org/apache/gossip/transport/http/SimpleNettyTransportManager.java

```diff
@@ -0,0 +1,268 @@
+    super(gossipManager, gossipCore);
+    this.endpointPort = gossipManager.getMyself().getUri().getPort();
+    endpointConsumer = new Thread("test-endpoint-consumer") {
+      public void run() {
+        while (!requestEndpointShutdown) {
+          try {
+            byte[] buf = gossipManager.getTransportManager().read();
+            try {
+              Base message = gossipManager.getProtocolManager().read(buf);
+              gossipCore.receive(message);
+              gossipManager.getMemberStateRefresher().run();
+            }
+            catch (RuntimeException ex) {
+              LOGGER.error("Unable to process message", ex);
+            }
+          } catch (IOException e) {
+            // InterruptedException are completely normal here because of the blocking lifecycle.
+            if (!(e.getCause() instanceof InterruptedException)) {
+              LOGGER.error(e);
+            }
+          }
+        }
+      }
+    };
+  }
+
+  @Override
+  public void startEndpoint() {
+    endpointConsumer.start();
+    server.start();  
+  }
+
+  @Override
+  public void send(URI endpoint, byte[] buf) throws IOException {
+    String base64Msg = Base64.getEncoder().encodeToString(buf);
+    new HttpClient(base64Msg)
+        .send(endpoint.getHost(), endpoint.getPort());
+  }
+
+  @Override
+  public byte[] read() throws IOException {
+    try {
+      byte[] msg = messages.take();
+      return msg;
+    } catch (InterruptedException ex) {
+      throw new IOException(ex);
+    }
+  }
+
+  @Override
+  public void shutdown() {
+    requestEndpointShutdown = true;
+    endpointConsumer.interrupt();
+    server.shutdown();
+    super.shutdown();
+  }
+
+  private class HttpClient {
+    private String message;
+    HttpClient(String message) {
+      this.message = message;
+    }
+    
+    public void send(String host, int port) {
+      EventLoopGroup group = new NioEventLoopGroup();
+      try {
+        Bootstrap b = new Bootstrap();
+        b.group(group)
+            .channel(NioSocketChannel.class)
+            .option(ChannelOption.TCP_NODELAY, true)
+            .handler(new ChannelInitializer<SocketChannel>() {
+              @Override
+              protected void initChannel(SocketChannel ch) throws Exception {
+                ChannelPipeline cp = ch.pipeline();
+                cp.addLast(new HttpClientCodec());
+                //cp.addLast(new HttpContentDecompressor());
+                //cp.addLast(new HttpObjectAggregator());
+                cp.addLast(new SimpleChannelInboundHandler<HttpObject>() {
+                  @Override
+                  protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Exception {
+                    if (msg instanceof HttpResponse) {
+                      HttpResponse res = (HttpResponse)msg; 
+//                      LOGGER.info("RESPONSE: " + res.getStatus());
+                    } else if (msg instanceof HttpContent) {
```

Comment:
No good reason. I'm not a netty expert. This code is adapted from what I've seen elsewhere and only exists to show what is currently required to create another `TransportManager`.